apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloClassGenTask.java

```diff
@@ -42,8 +42,11 @@ void generateClasses(IncrementalTaskInputs inputs) {
     inputs.outOfDate(new Action<InputFileDetails>() {
       @Override
       public void execute(InputFileDetails inputFileDetails) {
+        boolean useOptional = (nullValueType != ApolloExtension.NullableValueType.annotated);
+        boolean useGuava = (nullValueType == ApolloExtension.NullableValueType.guavaOptional
+            || (hasGuavaDep && nullValueType == ApolloExtension.NullableValueType.autoOptional));
```

Comment:
Might as well get rid of the `autoOptional` and the `useOptional` boolean and only have `annotated`, `guavaOptional`, `apolloOptional` be part of the enum.

Less configuration for the user and it's safe to assume that someone doing `ApolloExtension.NullableValueType.guavaOptional` would have guava in classpath.